models/core/logistic_regression.py

```diff
@@ -0,0 +1,115 @@
+from sklearn import linear_model
+import pandas as pd
+import joblib
+import os
+from data.datasets import base_loader
+from data import data_loader
+from data.adapters import continuous_adapter, categorical_adapter
+from data import recourse_adapter
+from models.core import model_trainer
+from models import model_constants, model_interface
+
+
+MODEL_FILENAME = "model.pkl"  # The default filename for saved LR models.
+TRAINING_PARAMS = {"class_weight": "balanced"}  # Params used for training.
+
+
+class LogisticRegression(model_trainer.ModelTrainer):
+    """A class for training, saving, and loading Logistic Regression models.
+
+    Uses the SKLearn LogisticRegression class."""
+
+    def __init__(
+        self,
+        dataset_name: data_loader.DatasetName,
+        model_name: model_constants.ModelName,
+    ):
+        """Creates a new LogisticRegression class.
+
+        Args:
+            dataset_name: The name of the dataset to load.
+            model_name: The name of the model to train."""
+        super().__init__(
+            model_type=model_constants.ModelType.LOGISTIC_REGRESSION,
+            dataset_name=dataset_name,
+            model_name=model_name,
+        )
+
+    def train_model(
+        self, dataset: pd.DataFrame, dataset_info: base_loader.DatasetInfo
+    ) -> model_interface.Model:
+        """Trains a model on the given training dataset.
+
+        Args:
+            dataset: The training dataset to use.
+            dataset_info: Information on the training dataset.
+
+        Returns:
+            A trained Model."""
+        adapter = self._get_adapter(dataset, dataset_info)
+        lr = linear_model.LogisticRegression(**TRAINING_PARAMS)
+        dataset = adapter.transform(dataset)
+        X = dataset.drop(dataset_info.label_name, axis=1)
```

Comment:
Better to use `training_data` and `labels` or other clearer names over `X` and `y`.

Reply:
At this stage in the project, focus on consistency. This is an important style and readability concern for the future.